src/Features/LanguageServer/Protocol/Handler/Symbols/WorkspaceSymbolsHandler.cs

```diff
@@ -17,18 +17,25 @@
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
     [Shared]
-    [ExportLspMethod(Methods.WorkspaceSymbolName)]
-    internal class WorkspaceSymbolsHandler : AbstractRequestHandler<WorkspaceSymbolParams, SymbolInformation[]>
+    [ExportLspMethod(Methods.WorkspaceSymbolName, mutatesSolutionState: false)]
+    internal class WorkspaceSymbolsHandler : IRequestHandler<WorkspaceSymbolParams, SymbolInformation[]>
     {
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public WorkspaceSymbolsHandler(ILspSolutionProvider solutionProvider) : base(solutionProvider)
+        public WorkspaceSymbolsHandler()
         {
         }
 
-        public override async Task<SymbolInformation[]> HandleRequestAsync(WorkspaceSymbolParams request, RequestContext context, CancellationToken cancellationToken)
+        public TextDocumentIdentifier? GetTextDocumentIdentifier(WorkspaceSymbolParams request) => null;
+
+        public async Task<SymbolInformation[]> HandleRequestAsync(WorkspaceSymbolParams request, RequestContext context, CancellationToken cancellationToken)
         {
             var solution = context.Solution;
+            if (solution == null)
```

Comment:
should we assert it's not null instead?  I would imagine if we get a workspace symbols request we should have some solution.

Reply:
Well spotted, this is actually old code, `solution` can't be null here any more.